src/test/java/guitests/TagPersonGuiTest.java

```diff
@@ -0,0 +1,276 @@
+package guitests;
+
+import address.model.datatypes.AddressBook;
+import guitests.guihandles.EditPersonDialogHandle;
+import guitests.guihandles.PersonCardHandle;
+import guitests.guihandles.TagPersonDialogHandle;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TagPersonGuiTest extends GuiTestBase {
+    @Override
+    protected AddressBook getInitialData() {
+        return td.book;
+    }
+
+    @Test
+    public void tagPerson() {
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+        EditPersonDialogHandle aliceEditDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        TagPersonDialogHandle aliceTagDialog = aliceEditDialog.openTagPersonDialogUsingShortcut();
+
+        aliceTagDialog.enterSearchQuery("coll").acceptSuggestedTag();
+        aliceTagDialog.close();
+
+        aliceEditDialog.pressEnter();
+
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+        sleepForGracePeriod();
+        assertEquals("Tag: colleagues", alicePersonCard.getTags());
+        assertFalse(alicePersonCard.isShowingGracePeriod("Editing"));
```

Comment:
Write a convenience method (in this class itself) to do these 5 steps? It can be used in many other places in this class.